cdk/src/functions/generate-mail-address.ts

```diff
@@ -1,132 +1,63 @@
-
-      parseAccountResponse(response.Accounts!);
-    }
-  }
 
-  return data;
-};
+async function getAccounts(): Promise<Account[]> {
+  const accounts: Account[] = [];
 
-const getSuspendedAccounts = async () => {
-  const data: TmpAccount[] = [];
 
   try {
-    await Organizations.listAccounts({}).promise();
+    const paginator = paginateListAccounts(
+      { client, pageSize: 20 }, {},
+    );
+
+    for await (const page of paginator) {
+      accounts.push(...page.Accounts!);
+    }
   } catch (e) {
-    // @ts-ignore
-    if (e.code == 'AWSOrganizationsNotInUseException') {
-      return data;
+    if (e instanceof AWSOrganizationsNotInUseException) {
+      return [];
     }
   }
 
-  const root = await Organizations.listRoots({}).promise();
-  const rootId = root.Roots![0].Id;
-
-  let ouList: AWS.Organizations.OrganizationalUnits = [];
-  let ouResponse;
-  try {
-    ouResponse = await Organizations.listOrganizationalUnitsForParent({
-      ParentId: rootId!,
-    }).promise();
-  } catch (e) {
-    return data;
-  }
+  return accounts;
+};
 
-  const parseOUResponse = (ous: AWS.Organizations.OrganizationalUnits) => {
-    ous.forEach((ou) => {
-      ouList.push(ou);
-    });
-  };
-
-  if (ouResponse) {
-    parseOUResponse(ouResponse.OrganizationalUnits!);
-    while (ouResponse.NextToken) {
-      ouResponse = await Organizations.listOrganizationalUnitsForParent({
-        ParentId: rootId!,
-        NextToken: ouResponse.NextToken,
-        MaxResults: 20,
-      }).promise();
-
-      parseOUResponse(ouResponse.OrganizationalUnits!);
-    }
-  }
+export function generateEmail(domain: string): string {
+  const maxCharacters = 64;
+  const availableCharacters = maxCharacters - (domain.length + 1 + 5); // root+{uuid}@domain.tld
+  const id = randomUUID().substring(0, availableCharacters);
 
-  const suspendedOU = ouList.filter(ou => ou.Name === 'Suspended');
+  const email = `root+${id}@${domain}`;
 
-  if (suspendedOU.length < 1) {
-    return data;
-  }
-  const suspendedOUId = suspendedOU[0].Id;
-
-  let accountsResponse = await Organizations.listAccountsForParent({ ParentId: suspendedOUId! }).promise();
-
-  const parseAccountResponse = (accounts: AWS.Organizations.Accounts) => {
-    accounts.forEach((account) => {
-      data.push({ id: account.Id!, name: account.Name!, email: account.Email! });
-    });
-  };
-
-  if (accountsResponse) {
-    parseAccountResponse(accountsResponse.Accounts!);
-    while (accountsResponse.NextToken) {
-      accountsResponse = await Organizations.listAccountsForParent({
-        ParentId: suspendedOUId!,
-        NextToken: accountsResponse.NextToken,
-        MaxResults: 20,
-      }).promise();
-
-      parseAccountResponse(accountsResponse.Accounts!);
-    }
+  if (email.length > 64) {
+    throw new Error('Unable to generate email address with more than 64 characters (Control Tower requirement)');
   }
 
-  return data;
-};
+  console.log('Created new email for account', email);
```

Comment:
not important .. but won't that result in the log info "Created new email for accountxample@sth.whatever"?

Reply:
No, according to the documentation there is automagically a space added.